### PR TITLE
WFLY-5112 Set map attribute to undefined if it is empty after remove …

### PR DESCRIPTION
**Pull request by tadamski**
…operation

### CODE REVIEW

File: controller/src/main/java/org/jboss/as/controller/operations/global/MapOperations.java [hunk 1]
```diff
@@ -155,6 +157,9 @@ private MapRemoveHandler() {
         void updateModel(final OperationContext context, ModelNode model, AttributeDefinition attributeDefinition, ModelNode attribute) throws OperationFailedException {
             String key = KEY.resolveModelAttribute(context, model).asString();
             attribute.remove(key);
+            if(attribute.asPropertyList().isEmpty()){
```
Comment (ctomc) [inline controller/src/main/java/org/jboss/as/controller/operations/global/MapOperations.java]: bit faster way would be attribute.asList() also would avoid some possible issues with type of model

**Comment by tadamski** [comment]
Fixed. I have also added ability to parse empty property list to PropertiesParser. Such configuration is redundant but technically valid IMO.

**Comment by ctomc** [comment]
:+1:

File: controller/src/main/java/org/jboss/as/controller/operations/global/MapOperations.java [hunk 2]
```diff
@@ -155,6 +157,9 @@ private MapRemoveHandler() {
         void updateModel(final OperationContext context, ModelNode model, AttributeDefinition attributeDefinition, ModelNode attribute) throws OperationFailedException {
             String key = KEY.resolveModelAttribute(context, model).asString();
             attribute.remove(key);
+            if(attribute.asList().isEmpty()){
+                attribute.set(UNDEFINED);
```
Comment (jamezp) [inline controller/src/main/java/org/jboss/as/controller/operations/global/MapOperations.java]: This should be `attribute.clear()`. The static `UNDEFINED` node is mutable.

**Comment by jamezp** [comment]
If this fixes the WFLY-5112 issue then the JIRA needs to be moved to WFCORE. If this is a separate fix then we need a new WFCORE PR.

**Comment by tadamski** [comment]
Fixed. I have created new https://issues.jboss.org/browse/WFCORE-894 for this PR.

**Comment by jamezp** [comment]
retest this please

**Comment by jamezp** [comment]
One more time retest this please

**Comment by bstansberry** [comment]
retest this please

The "Core - Full Integration" failure looks like the pattern we've seen before when some zombie process was sitting around on a CI slave.

**Pull request merged**